src/interface/TransactionRequestsService-swagger.yaml

```diff
@@ -987,8 +987,7 @@ components:
     AuthenticationValue:
       title: AuthenticationValue
       oneOf:
-        - $ref: '#/components/schemas/OtpValue'
-        - $ref: '#/components/schemas/QRCODE'
+        - $ref: '#/components/schemas/QRCODEorOTPValue'
         - $ref: '#/components/schemas/U2FPinValue'
```

Comment:
Something about this way of handling irks me.
Like if I was to hypothetically add more auth methods that were strings it would be
`QRCODEorOTPValueorNewAuth1orNewAuth2`.

Maybe just let all auth methods be objects even if they only contain one key/value?

Anyone want to chime in?

Reply:
For oneOf there is requirements that all variants have to be separate (in matching sense) so if the new request is validating and more than one variant match is found the 400 status is reported. So I have had to merge overlapping regexp cases. We cant go for object for all cases because we have to be back compatible

Reply:
Maybe putting multiple optional fields for different cases would be a solution... is it worth to refactor whole flow for such a change?

Reply:
Hmm. I'm ok with this for now - it's kinda ugly, but that's one of the annoying things about trying to express logic in open api definitions that we might just have to live with. 

@eoln is the `oneOf` requirement you are talking about at an OAS level? Or is it just an implementation quirk of the libraries we are using?

Reply:
@lewisdaly I didn't find anything about in OAS. I don't think it is a quirk. The string value of `AuthenticationValue` has overlapping regexps for QRCODE(\S{1,64}) and OTP (\d{3,10}): so the value '123456' match both of them and the conflict appears so the validation decision should be made: which schema is valid?  I think it is a good decision to set 400 status - It look weird for a string field, but if you have two variant schemas of type object with properties secured by regexp, and for both schemas, one property field is matching the request then what? I remember than at the beginning of PI I've seen OTP & QRCODE regexps merged - then we migrated to oneOf version and everything was okay until yesterday when I was implementing unit tests and for the case as above I got validation failure.

Reply:
Do you have any opinion about @rmothilal ?

Reply:
@elon could anyOf not be used instead of oneOf?

Reply:
@rmothail it was worth waiting your opinion! Let me try!